apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/block-iframe-preview.js

```diff
@@ -121,6 +122,24 @@ const BlockFramePreview = ( {
 		} );
 	}, [ viewportWidth ] );
 
+	// Set template title.
+	useEffect( () => {
+		const iframeBody = get( iframeRef, [ 'current', 'contentDocument', 'body' ] );
+		if ( ! iframeBody ) {
+			return;
+		}
+
+		const templateTitle = iframeBody.querySelector(
+			'.editor-post-title .editor-post-title__input'
+		);
+
+		if ( ! templateTitle ) {
+			return;
+		}
+
+		templateTitle.value = title;
```

Comment:
Minor: I think a quick comment here to explain that this is a workaround solution and why it's required.

I think you [had this originally and I suggested some changes](https://github.com/Automattic/wp-calypso/pull/39831#discussion_r387558917), but it's been lost in your subsequent fixes.